tests/utests/node/list.c

```diff
@@ -0,0 +1,1670 @@
+            "   {\"uid\":1, \"name\":\"Antonín Kuba\", \"group\":\"User\"},"
+            "   {\"uid\":2, \"name\":\"Antonín Kuba\", \"group\":\"User\"},"
+            "   {\"uid\":3, \"name\":\"Antonín Kuba\", \"group\":\"User\"},"
+            "   {\"uid\":4, \"name\":\"Tomáš Novák\", \"group\":\"Admin\"}"
+            "]}";
+    CHECK_PARSE_LYD_PARAM(data, LYD_JSON, 0, LYD_VALIDATE_PRESENT, LY_SUCCESS, tree);
+    list_tree = (void *)tree;
+    CHECK_LYD_NODE_INNER(list_tree, 1, 0, 0, 0, 1, 0, 0, 1);
+    list_tree = (void *) list_tree->next;
+    CHECK_LYD_NODE_INNER(list_tree, 1, 0, 0, 0, 1, 0, 0, 1);
+    list_tree = (void *) list_tree->next;
+    CHECK_LYD_NODE_INNER(list_tree, 1, 0, 0, 0, 1, 0, 0, 1);
+    list_tree = (void *) list_tree->next;
+    CHECK_LYD_NODE_INNER(list_tree, 1, 0, 0, 0, 1, 0, 0, 1);
+    list_tree = (void *) list_tree->next;
+    CHECK_LYD_NODE_INNER(list_tree, 1, 0, 0, 0, 0, 0, 0, 1);
+    lyd_free_all(tree);
+
+    /* check wrong number of items */
+    data =
+            "{\"T2:user\": ["
+            "   {\"uid\":0, \"name\":\"Jan Kuba\", \"group\":\"User\"},"
+            "   {\"uid\":4, \"name\":\"Tomáš Novák\", \"group\":\"Admin\"}"
+            "]}";
+    CHECK_PARSE_LYD_PARAM(data, LYD_JSON, 0, LYD_VALIDATE_PRESENT, LY_EVALID, tree);
+    assert_null(tree);
+    CHECK_LOG_CTX("Too few \"user\" instances.",
+            "Schema location /T2:user, line number 1.");
+
+    data =
+            "{\"T2:user\": ["
+            "   {\"uid\":0, \"name\":\"Jan Kuba\", \"group\":\"User\"},"
+            "   {\"uid\":1, \"name\":\"Antonín Kuba\", \"group\":\"User\"},"
+            "   {\"uid\":2, \"name\":\"Antonín Kuba\", \"group\":\"User\"},"
+            "   {\"uid\":3, \"name\":\"Antonín Kuba\", \"group\":\"User\"},"
+            "   {\"uid\":4, \"name\":\"Tomáš Novák\", \"group\":\"Admin\"},"
+            "   {\"uid\":5, \"name\":\"Tomáš Novák\", \"group\":\"Admin\"}"
+            "]}";
+    CHECK_PARSE_LYD_PARAM(data, LYD_JSON, 0, LYD_VALIDATE_PRESENT, LY_EVALID, tree);
+    assert_null(tree);
+    CHECK_LOG_CTX("Too many \"user\" instances.",
+            "Schema location /T2:user, data location /T2:user[uid='5'], line number 1.");
+
+    schema = MODULE_CREATE_YANG("T_EMPTY_LIST",
+            "container user_list {"
+            "   list user {"
+            "   key uid;"
+            "   unique \"name group\";"
+            "   leaf uid{type uint32;}"
+            "   leaf name{type string;}"
+            "   leaf group{type string;}"
+            "}"
+            "}");
+    UTEST_ADD_MODULE(schema, LYS_IN_YANG, NULL, NULL);
+
+    /* empty list */
+    data =
+            "{\"T_EMPTY_LIST:user_list\": {}"
+            "}";
+
+    /* check first item */
+    CHECK_PARSE_LYD_PARAM(data, LYD_JSON, 0, LYD_VALIDATE_PRESENT, LY_SUCCESS, tree);
+    list_tree = (void *)tree;
+    CHECK_LYD_NODE_INNER(list_tree, 0, 0, 0, 1, 0, 0, 0, 1);
+    lyd_free_all(tree);
+
+    /* DOESNT WORK
```

Comment:
Yes, you simply cannot create a list instance without its keys, so this can be removed, I suppose.